### DISCUSSION TITLE
Replicating an in memory dB to another app instance

### DISCUSSION THREAD

**Discussion by StephenFlavin** [duckdb/duckdb#13246] [Q&A] [answered]
Hi,

I'm trying to build an API application in java that uses an in memory duckdb database backed by a compact Kafka topic which allows the app to scale completely horizontally with very low latency data access (plan to benchmark against other embedded DBs)

This API is for fun at the moment so I'm trying to get all the performance I can out of it and given horizontal scaling is the goal I wondered if there was a way to replicate an existing instances DB via an API call? Only then using the compact Kafka topic as a fallback to get a faster startup time.

So far best I can come up with is generating a csv with `select * from foo` though I'm also not sure how to use copy from a csv that never gets written to disk.

Any thoughts welcomed 🙂

**Comment by Tishj** [comment, marked as answer]
EXPORT/IMPORT DATABASE sound like your only options

A more sane way to do this is using a persistent database as that gets properly serialized/deserialized

**Reply by StephenFlavin** [reply]
Thanks I'll try that out, thought I was missing something with in memory dbs
